File: scala/sbt/emr/flink-starter-project-aws-managed-flink/src/main/scala/org/example/connectors/sink/ConsoleSinkFunction.scala
package org.example.connectors.sink

import org.apache.flink.streaming.api.functions.sink.{RichSinkFunction, SinkFunction}
import org.slf4j.{Logger, LoggerFactory}

class ConsoleSinkFunction extends RichSinkFunction[String] {
  private val logger: Logger = LoggerFactory.getLogger(getClass)

  @throws[Exception]
  override def invoke(value: String, context: SinkFunction.Context): Unit = {
    logger.info("output: {}", value)
  }
}

File: java/maven/emr/flink-started-project/src/main/java/org/example/flink/skeleton/connections/sinks/ConsoleSinkFunction.java
package org.example.flink.skeleton.connections.sinks;

import lombok.Builder;
import lombok.extern.slf4j.Slf4j;
import org.apache.flink.streaming.api.functions.sink.RichSinkFunction;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Builder
@Slf4j
public class ConsoleSinkFunction extends RichSinkFunction<String> {
    @Override
    public void invoke(String value, Context context) throws Exception {
        log.info("output:{}", value);
    }
}
